### ISSUE TITLE
Dune version constraint in the opam file

### ISSUE THREAD

**Issue by jeremiedimino** [ocaml/dune#1689]
When one has `(lang dune 1.4)` in their opam file, then they must also have `"dune" {>= "1.4"}` in their opam file. Users might easily forget to update this constraint when bumping the dune lang version.

Dune could help by warning when it detects that the constraint is wrong. Additionally, when generating the opam file, dune should produce the right lower-bound (/cc @avsm).

**Comment by emillon** [comment]
Once there is a stable representation built for `dune describe`, we might be able to do some lightweight checks on it. This could be one of these checks.

**Comment by Et7f3** [comment]
Update: Now if w use the `generate_opam_files` feature dune is automatically added with the right version but if we add dune (explicitly in depends field) with wrong lower-bound version then we can have ill-formed opam package. So just the check isn't done and warning not emitted.

dune: 2.5.1

```
(lang dune 2.2)

(generate_opam_files true)

(package
 (name foo)
 (depends
  (dune (<= 2.0))))
```

generate:
```
# This file is generated by dune, edit dune-project instead
opam-version: "2.0"
depends: [
  "dune" {<= "2.0"}
]
build: [
  ["dune" "subst"] {pinned}
  [
    "dune"
    "build"
    "-p"
    name
    "-j"
    jobs
    "@install"
    "@runtest" {with-test}
    "@doc" {with-doc}
  ]
]
```

**Comment by jeremiedimino** [comment]
@Et7f3 this problem (and the original one) should be fixed by #3434 which will be part of the next dune release

**Comment by Et7f3** [comment]
We can still create illegal constraint without receiving warning that's why I have opened #3468
